public/app/plugins/panel/heatmap/rendering.ts

```diff
@@ -366,11 +367,18 @@ export class HeatmapRenderer {
 
     const yAxis = d3
       .axisLeft(this.yScale)
-      .tickValues(tickValues)
       .tickFormat(tickFormatter)
       .tickSizeInner(0 - this.width)
       .tickSizeOuter(0)
       .tickPadding(Y_AXIS_TICK_PADDING);
+    if (
+      (tickValues && tickValues.length <= ticks) ||
+      (this.panel.yBucketBound === 'middle' && tickValues && tickValues.length)
+    ) {
```

Comment:
Why do we need to leave `middle` ticks as is? From my POV there's no difference in terms of how to display it.

```suggestion
    if (tickValues && tickValues.length <= ticks) {
```

Reply:
I was simply trying to preserve pre-existing functionality. I have no objection to this change.